### ISSUE TITLE
[Feature request] Pass item text to AutoLink's before and after slots

### ISSUE THREAD

**Issue by stfsy** [vuepress/vuepress-next#896]
### Clear and concise description of the problem

I'm publishing OpenAPI docs with VuePress. The sidebar of the docs contain resource and endpoint name. The structure is as follows:

\# API Reference (static)
\## << Resource name >>
\### << HTTP Verb >> << Endpoint name>> 

![image](https://user-images.githubusercontent.com/2894615/168467478-5d3d5928-bb5c-45c9-88de-2ab2bee3ccb7.png)

As you can see on the image above I added `Badges` to the Endpoint names, because (a) that's typically done by other OpenAPI UIs too (see e.g. https://swagger.io/tools/swagger-ui/) and (b) it's easier to distinguish the endpoints.

The challenge is the `SideBar` item components do not render Vue components that are part of the item text. That is because `AutoLink` does not render the HTML, which I think is fine.

### Suggested solution

However, the `AutoLink` component already defines [slots](https://github.com/vuepress/vuepress-next/blob/main/packages/%40vuepress/theme-default/src/client/components/AutoLink.vue#L95). Two for internal and two for external links. 

If I pass the item text to the templates, I can easily add the Badges to the _before_ slots without changing the actual `AutoLink` component.

### AutoLink (VuePress built-in)
```vue
<template>
  <RouterLink v-if="isRouterLink" :class="{ 'router-link-active': isActive }" :to="item.link"
    :aria-label="linkAriaLabel" v-bind="$attrs">

    <slot name="before" :text="item.text" />
    {{ item.text }}
    <slot name="after" :text="item.text" />
  </RouterLink>

  <a v-else class="external-link" :href="item.link" :rel="linkRel" :target="linkTarget" :aria-label="linkAriaLabel"
    v-bind="$attrs">
    <slot name="before" :text="item.text" />
    {{ item.text }}
    <AutoLinkExternalIcon v-if="isBlankTarget" />
    <slot name="after" :text="item.text" />
  </a>
</template>
```

Which allows me to add this AutoLinkHttpEndpoint:

### AutoLinkHttpEndpoint (Project-specific)
```vue
<script setup lang="ts">
import ParentLayout from './AutoLink.vue'

const props = defineProps({
  item: {
    type: Object as PropType<NavLink>,
    required: true,
  },
})
</script>

<template>
  <ParentLayout :item="item">
    <template #before="props">

      <Badge v-if="props.text.startsWith('Get ')" text="GET" type="get" vertical="middle" />
      <Badge v-else-if="props.text.startsWith('Create')" text="POST" type="post" vertical="middle" />
      <Badge v-else-if="props.text.startsWith('Add')" text="POST" type="post" vertical="middle" />
      <Badge v-else-if="props.text.startsWith('Update')" text="PUT" type="put" vertical="middle" />
      <Badge v-else-if="props.text.startsWith('Delete')" text="DELETE" type="delete" vertical="middle" />

    </template>
  </ParentLayout>
</template>

```

### Alternative

Well I could override the AutoLink component. Though I feel we could provide value to other developers and teams too by passing the item text to the slots.

**Comment by Mister-Hope** [comment]
I think that even you want to modify it, it should be like this way:

```diff
<template>
  <RouterLink v-if="isRouterLink" :class="{ 'router-link-active': isActive }" :to="item.link"
    :aria-label="linkAriaLabel" v-bind="$attrs">

-    <slot name="before" :text="item.text" />
+    <slot name="before" v-bind="item" />
    {{ item.text }}
-     <slot name="after" :text="item.text" />
+    <slot name="after" v-bind="item" />
  </RouterLink>

  <a v-else class="external-link" :href="item.link" :rel="linkRel" :target="linkTarget" :aria-label="linkAriaLabel"
    v-bind="$attrs">
    <slot name="before" :text="item.text" />
    {{ item.text }}
    <AutoLinkExternalIcon v-if="isBlankTarget" />
    <slot name="after" :text="item.text" />
  </a>
</template>
```

Also, you should be able to overide AutoLink with alias. I feel ok both currect and the above requested. Fell free to open a pr.

**Comment by stfsy** [comment]
@Mister-Hope Thanks for having a look already and thank you for the kind response. 👍 

I already started creating a PR and after reading your comment I realized my custom component `AutoLinkHttpEndpoint` already has access to the item and the item's text. That's because the custom component needs to define the item as a prop too. So the item text was actually already in scope. 

From my perspective it's not necessary to create a scoped slot then. Let's keep it simple. 👌
